internal/kafka/kafka.go

```diff
@@ -76,30 +76,34 @@ func InitializeKafkaBroker(ctx context.Context) error {
 }
 
 func NewKafkaBroker(ctx context.Context) (Broker, error) {
+	var pool *x509.CertPool
 	var tlsConfig *tls.Config
 	var saslMechanism sasl.Mechanism
 
 	logger := zerolog.Ctx(ctx)
-	logger.Debug().Msgf("Setting up Kafka transport: %v CA:%v SASL:%v", config.Kafka.Brokers,
-		config.Kafka.CACert != "", config.Kafka.SASL.SaslMechanism != "" && config.Kafka.SASL.SaslMechanism != "none")
+	logger.Debug().Msgf("Setting up Kafka transport: %v", config.Kafka.Brokers)
 
-	// configure TLS when CA certificate was provided
 	if config.Kafka.CACert != "" {
-		logger.Debug().Str("cert", config.Kafka.CACert).Msg("Adding CA certificates to the pool")
+		logger.Debug().Str("cert", config.Kafka.CACert).Msg("Configuring TLS CA pool for Kafka")
 
 		pemCerts := config.Kafka.CACert
-		pool := x509.NewCertPool()
+		pool = x509.NewCertPool()
 		if ok := pool.AppendCertsFromPEM([]byte(pemCerts)); !ok {
 			logger.Warn().Msg("Could not add an CA cert to the pool")
 		}
+	}
+
+	if config.Kafka.TlsEnabled && !config.InEphemeralClowder() {
```

Comment:
Isn't there a way to figure out wether we have TLS from clowder config?

I guess we can do some follow-ups, if this passes, lets merge as-is, but it looks bit suspicius to me :)

Reply:
If this works, we can remove this line and properly disable TLS in app-interface for ephemeral. But I just want to know if this actually works or not.